cw-orch-daemon/src/tx_builder.rs

```diff
@@ -67,8 +67,17 @@ impl TxBuilder {
     }
 
     pub(crate) fn build_fee(amount: impl Into<u128>, denom: &str, gas_limit: u64) -> Fee {
-        let fee = Coin::new(amount.into(), denom).unwrap();
-        Fee::from_amount_and_gas(fee, gas_limit)
+        // Ensure that the fee is not 0, which can be invalid
+        let fee = match amount.into() {
+            0 => vec![],
+            x => vec![Coin::new(x, denom).unwrap()],
+        };
+        Fee {
+            gas_limit,
+            amount: fee,
+            payer: None,
+            granter: None
+        }
```

Comment:
This one was really interesting - because in simulation, it would build with 0 tokens as the fee, which to some nodes is invalid.

Reply:
Nice catch